modules/openapi-generator/src/main/resources/php/ObjectSerializer.mustache

```diff
@@ -315,8 +315,9 @@ class ObjectSerializer
                     // Some API's return a date-time with too high nanosecond
                     // precision for php's DateTime to handle. This conversion
                     // (string -> unix timestamp -> DateTime) is a workaround
-                    // for the problem.
-                    return (new \DateTime())->setTimestamp(strtotime($data));
+                    // for the problem. Note that strtotime only handles precision
+                    // up to a second, so trimming does not result in data loss.
+                    return (new \DateTime())->setTimestamp(strtotime(substr($data, 0, strpos($data, '.'))));
```

Comment:
@carmenquan thanks for the PR. What if $data doesn't contain `.` (e.g. `2021-10-06T20:17:16`)? Would the fix still work?

Reply:
This fix cuts off timezone part, which is really wrong. I used regexp instead and added tests to cover that issue. Just cannot push my additional commits to the author's fork.

Reply:
@ybelenko I See. Can you please file a new PR instead?